Cannot start the Loba-chat in Ubuntu 24.04.2 and Linux Mint 22.1 for pglite problem

### 📦 Platform

Other

### 📦 Deploymenet mode

client db (lobe-chat image)

### 📌 Version

All of Recent releases version

### 💻 Operating System

Ubuntu

### 🌐 Browser

Edge

### 🐛 Bug Description

When try to start  the Loba-chat Appimage (get from release page) from terminal, it would shows the 'request failed' error.


In terminal, logs:

```
 /ExtDisk//SW/LobeHub-Beta-1.107.3.AppImage 
(node:3052124) ExperimentalWarning: SQLite is an experimental feature and might change at any time
(Use `lobehub-desktop-beta --trace-warnings ...` to show where the warning was created)
(node:3052124) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
Warning: Cannot load "@napi-rs/canvas" package: "Error: Cannot find module '@napi-rs/canvas'
Require stack:
- /home/runner/work/lobe-chat/lobe-chat/apps/desktop/node_modules/.pnpm/pdfjs-dist@4.10.38/node_modules/pdfjs-dist/legacy/build/pdf.mjs".
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":false,"isInitializing":false}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Database path: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db
⚠️ Another process has already locked the database: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db.lock
❌ Failed to initialize PGlite instance: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
{"level":30,"time":1754272138684,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: agent.getAgentConfig, type: query"}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
{"level":30,"time":1754272138685,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: aiProvider.getAiProviderRuntimeState, type: query"}
{"level":30,"time":1754272138685,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: user.getUserState, type: query"}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":false,"isInitializing":false}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Database path: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db
⚠️ Another process has already locked the database: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db.lock
❌ Failed to initialize PGlite instance: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
{"level":30,"time":1754272138719,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: agent.getAgentConfig, type: query"}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
{"level":30,"time":1754272138719,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: aiProvider.getAiProviderRuntimeState, type: query"}
{"level":30,"time":1754272138719,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: user.getUserState, type: query"}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":false,"isInitializing":false}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Database path: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db
⚠️ Another process has already locked the database: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db.lock
❌ Failed to initialize PGlite instance: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
{"level":30,"time":1754272139064,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: session.getGroupedSessions, type: query"}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
{"level":30,"time":1754272139064,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: plugin.getPlugins, type: query"}
{"level":30,"time":1754272139064,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: agent.getAgentConfig, type: query"}
{"level":30,"time":1754272139064,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: topic.getTopics, type: query"}
{"level":30,"time":1754272139064,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: message.getMessages, type: query"}
Failed to update prerender cache for b3872802d4f50c61c6c517937fb4a395fa84c8cf4845b9e729c911f70f005eea [Error: ENOTDIR: not a directory, mkdir '/tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/cache/fetch-cache'] {
  errno: -20,
  code: 'ENOTDIR',
  syscall: 'mkdir',
  path: '/tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/cache/fetch-cache'
}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":false,"isInitializing":false}
Database path: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db
⚠️ Another process has already locked the database: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db.lock
❌ Failed to initialize PGlite instance: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
{"level":30,"time":1754272143707,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: agent.getAgentConfig, type: query"}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
      at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
      at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
      at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
      at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
      at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
      at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
      at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":false,"isInitializing":false}
Database path: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db
⚠️ Another process has already locked the database: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db.lock
❌ Failed to initialize PGlite instance: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
{"level":30,"time":1754272143751,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: aiProvider.getAiProviderRuntimeState, type: query"}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
      at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
      at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
      at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
      at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
      at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
      at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
      at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":false,"isInitializing":false}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Database path: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db
⚠️ Another process has already locked the database: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db.lock
❌ Failed to initialize PGlite instance: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
{"level":30,"time":1754272144177,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: plugin.getPlugins, type: query"}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
}
{"level":30,"time":1754272144177,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: agent.getAgentConfig, type: query"}
{"level":30,"time":1754272144177,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: topic.getTopics, type: query"}
{"level":30,"time":1754272144177,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: message.getMessages, type: query"}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":false,"isInitializing":false}
Database path: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db
⚠️ Another process has already locked the database: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db.lock
❌ Failed to initialize PGlite instance: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
{"level":30,"time":1754272148711,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: user.getUserState, type: query"}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18) {
  code: 'INTERNAL_SERVER_ERROR',
  [cause]: Error: Cannot acquire database lock. Another instance might be using it.
      at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
      at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
      at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
      at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
      at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
      at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
      at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
      at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":false,"isInitializing":false}
Database path: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db
⚠️ Another process has already locked the database: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db.lock
❌ Failed to initialize PGlite instance: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
❌ Failed to initialize database: Error: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18)
{"level":30,"time":1754272148729,"pid":3052124,"hostname":"XXXX-9950","msg":"Error in tRPC handler (lambda) on path: aiProvider.getAiProviderRuntimeState, type: query"}
Error [TRPCError]: Cannot acquire database lock. Another instance might be using it.
    at <unknown> (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:1007:27)
    at async getServerDB (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/67223.js:2966:86)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/app/(backend)/trpc/desktop/[trpc]/route.js:196:22)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async callRecursive (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1283:18)
    at async procedure (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/68967.js:1308:18)
    at async (../../../../../tmp/.mount_LobeHuS6bhHw/resources/app.asar/dist/next/.next/server/chunks/25071.js:1953:18) {
  code: 'INTERNAL_SERVER_ERROR',
```

Even if I manually remove the  ~/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db file, the issue still exists.


### 📷 Recurrence Steps

_No response_

### 🚦 Expected Behavior

_No response_

### 📝 Additional Information

_No response_

After rm the db.lock file, it now would crash and exit:

```
(node:3305169) ExperimentalWarning: SQLite is an experimental feature and might change at any time
(Use `lobehub-desktop-beta --trace-warnings ...` to show where the warning was created)
(node:3305169) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
Warning: Cannot load "@napi-rs/canvas" package: "Error: Cannot find module '@napi-rs/canvas'
Require stack:
- /home/runner/work/lobe-chat/lobe-chat/apps/desktop/node_modules/.pnpm/pdfjs-dist@4.10.38/node_modules/pdfjs-dist/legacy/build/pdf.mjs".
[3305262:0804/173110.042488:ERROR:gpu/command_buffer/service/shared_image/shared_image_manager.cc:389] SharedImageManager::ProduceSkia: Trying to Produce a Skia representation from a non-existent mailbox.
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":false,"isInitializing":false}
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Getting PGlite instance, state: {"hasExistingDB":false,"hasPromise":true,"isInitializing":true}
Waiting for existing initialization promise to complete
Database path: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db
✅ Successfully acquired database lock: /home/XXXX/.config/LobeHub-Beta/lobehub-storage/lobehub-local-db.lock
Creating new PGlite instance
PGlite state: true
schemaHash: 5f5693717af71ca191982a12d8247d4c
🚀 Starting Electron DB migration...
✅ Electron DB migration success, took 12ms
✅ PGlite instance successfully initialized
Failed to update prerender cache for b3872802d4f50c61c6c517937fb4a395fa84c8cf4845b9e729c911f70f005eea [Error: ENOTDIR: not a directory, mkdir '/tmp/.mount_LobeHuDoOxMO/resources/app.asar/dist/next/.next/cache/fetch-cache'] {
  errno: -20,
  code: 'ENOTDIR',
  syscall: 'mkdir',
  path: '/tmp/.mount_LobeHuDoOxMO/resources/app.asar/dist/next/.next/cache/fetch-cache'
}
[3305169:0804/173112.146822:ERROR:content/browser/browser_main_loop.cc:278] GLib-GObject: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
[3305169:0804/173112.146823:ERROR:content/browser/browser_main_loop.cc:278] GLib-GObject: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
[3305169:0804/173112.146825:ERROR:content/browser/browser_main_loop.cc:278] GLib-GObject: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
[3305169:0804/173112.146840:ERROR:content/browser/browser_main_loop.cc:278] GLib-GObject: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
[3305169:0804/173112.146840:ERROR:content/browser/browser_main_loop.cc:278] GLib-GObject: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
[3305169:0804/173112.146840:ERROR:content/browser/browser_main_loop.cc:278] GLib-GObject: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
[3305169:0804/173112.146852:ERROR:content/browser/browser_main_loop.cc:276] GLib-GObject: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
[3305169:0804/173112.146852:ERROR:content/browser/browser_main_loop.cc:276] GLib-GObject: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
[3305169:0804/173112.146852:ERROR:content/browser/browser_main_loop.cc:276] GLib-GObject: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
[3305169:0804/173112.146855:ERROR:content/browser/browser_main_loop.cc:276] GLib-GObject: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
[3305169:0804/173112.146857:ERROR:content/browser/browser_main_loop.cc:276] GLib-GObject: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
Trace/breakpoint trap (core dumped)
```

--- a/apps/desktop/src/main/core/App.ts
+++ b/apps/desktop/src/main/core/App.ts
@@ -1,11 +1,12 @@
 import { ElectronIPCEventHandler, ElectronIPCServer } from '@lobechat/electron-server-ipc';
 import { Session, app, ipcMain, protocol } from 'electron';
 import { macOS, windows } from 'electron-is';
+import { pathExistsSync, remove } from 'fs-extra';
 import os from 'node:os';
 import { join } from 'node:path';
 
 import { name } from '@/../../package.json';
-import { buildDir, nextStandaloneDir } from '@/const/dir';
+import { buildDir, LOCAL_DATABASE_DIR, nextStandaloneDir } from '@/const/dir';
 import { isDev } from '@/const/env';
 import { IControlModule } from '@/controllers';
 import { IServiceModule } from '@/services';
@@ -128,6 +129,9 @@ export class App {
     }
 
     this.initDevBranding();
+
+    // Clean up stale database lock file before starting IPC server
+    await this.cleanupDatabaseLock();
 
     //  ==============
     await this.ipcServer.start();
@@ -368,6 +372,27 @@ export class App {
     app.setName('lobehub-desktop-dev');
     if (macOS()) {
       app.dock!.setIcon(join(buildDir, 'icon-dev.png'));
+    }
+  };
+
+  /**
+   * Clean up stale database lock file from previous crashes or abnormal exits
+   */
+  private cleanupDatabaseLock = async () => {
+    try {
+      const dbPath = join(this.appStoragePath, LOCAL_DATABASE_DIR);
+      const lockPath = `${dbPath}.lock`;
+
+      if (pathExistsSync(lockPath)) {
+        logger.info(`Cleaning up stale database lock file: ${lockPath}`);
+        await remove(lockPath);
+        logger.info('Database lock file removed successfully');
+      } else {
+        logger.debug('No database lock file found, skipping cleanup');
+      }
+    } catch (error) {
+      logger.error('Failed to cleanup database lock file:', error);
+      // Non-fatal error, allow application to continue
     }
   };
 

--- a/apps/desktop/src/main/core/__tests__/App.test.ts
+++ b/apps/desktop/src/main/core/__tests__/App.test.ts
@@ -1,0 +1,282 @@
+import { app } from 'electron';
+import { pathExistsSync, remove } from 'fs-extra';
+import { join } from 'node:path';
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
+
+import { LOCAL_DATABASE_DIR } from '@/const/dir';
+
+// Mock electron modules
+vi.mock('electron', () => ({
+  app: {
+    getAppPath: vi.fn(() => '/mock/app/path'),
+    getLocale: vi.fn(() => 'en-US'),
+    getPath: vi.fn(() => '/mock/user/path'),
+    requestSingleInstanceLock: vi.fn(() => true),
+    whenReady: vi.fn(() => Promise.resolve()),
+    on: vi.fn(),
+    commandLine: {
+      appendSwitch: vi.fn(),
+    },
+    dock: {
+      setIcon: vi.fn(),
+    },
+    exit: vi.fn(),
+  },
+  ipcMain: {
+    handle: vi.fn(),
+  },
+  nativeTheme: {
+    on: vi.fn(),
+    shouldUseDarkColors: false,
+  },
+  protocol: {
+    registerSchemesAsPrivileged: vi.fn(),
+  },
+}));
+
+// Mock logger
+vi.mock('@/utils/logger', () => ({
+  createLogger: () => ({
+    debug: vi.fn(),
+    info: vi.fn(),
+    warn: vi.fn(),
+    error: vi.fn(),
+  }),
+}));
+
+// Mock fs-extra module
+vi.mock('fs-extra', async () => {
+  const actual = await vi.importActual('fs-extra');
+  return {
+    ...actual,
+    pathExistsSync: vi.fn(),
+    remove: vi.fn(),
+  };
+});
+
+// Mock common/routes
+vi.mock('~common/routes', () => ({
+  findMatchingRoute: vi.fn(),
+  extractSubPath: vi.fn(),
+}));
+
+// Mock other dependencies
+vi.mock('electron-is', () => ({
+  macOS: vi.fn(() => false),
+  windows: vi.fn(() => false),
+}));
+
+vi.mock('fix-path', () => ({
+  default: vi.fn(),
+}));
+
+vi.mock('@/const/env', () => ({
+  isDev: false,
+}));
+
+vi.mock('@/const/dir', () => ({
+  buildDir: '/mock/build',
+  nextStandaloneDir: '/mock/standalone',
+  LOCAL_DATABASE_DIR: 'lobehub-local-db',
+  appStorageDir: '/mock/storage/path',
+  userDataDir: '/mock/user/data',
+  DB_SCHEMA_HASH_FILENAME: 'lobehub-local-db-schema-hash',
+  FILE_STORAGE_DIR: 'file-storage',
+  INSTALL_PLUGINS_DIR: 'plugins',
+  LOCAL_STORAGE_URL_PREFIX: '/lobe-desktop-file',
+}));
+
+vi.mock('@lobechat/electron-server-ipc', () => ({
+  ElectronIPCServer: vi.fn().mockImplementation(() => ({
+    start: vi.fn().mockResolvedValue(undefined),
+  })),
+}));
+
+// Mock all infrastructure managers
+vi.mock('../infrastructure/I18nManager', () => ({
+  I18nManager: vi.fn().mockImplementation(() => ({
+    init: vi.fn().mockResolvedValue(undefined),
+  })),
+}));
+
+vi.mock('../infrastructure/StoreManager', () => ({
+  StoreManager: vi.fn().mockImplementation(() => ({
+    get: vi.fn((key) => {
+      if (key === 'storagePath') return '/mock/storage/path';
+      return undefined;
+    }),
+    set: vi.fn(),
+  })),
+}));
+
+vi.mock('../infrastructure/StaticFileServerManager', () => ({
+  StaticFileServerManager: vi.fn().mockImplementation(() => ({
+    initialize: vi.fn().mockResolvedValue(undefined),
+    destroy: vi.fn(),
+  })),
+}));
+
+vi.mock('../infrastructure/UpdaterManager', () => ({
+  UpdaterManager: vi.fn().mockImplementation(() => ({
+    initialize: vi.fn().mockResolvedValue(undefined),
+  })),
+}));
+
+vi.mock('../infrastructure/ProtocolManager', () => ({
+  ProtocolManager: vi.fn().mockImplementation(() => ({
+    initialize: vi.fn(),
+    processPendingUrls: vi.fn().mockResolvedValue(undefined),
+  })),
+}));
+
+vi.mock('../browser/BrowserManager', () => ({
+  BrowserManager: vi.fn().mockImplementation(() => ({
+    initializeBrowsers: vi.fn(),
+    getIdentifierByWebContents: vi.fn(),
+  })),
+}));
+
+vi.mock('../ui/MenuManager', () => ({
+  MenuManager: vi.fn().mockImplementation(() => ({
+    initialize: vi.fn(),
+  })),
+}));
+
+vi.mock('../ui/ShortcutManager', () => ({
+  ShortcutManager: vi.fn().mockImplementation(() => ({
+    initialize: vi.fn(),
+  })),
+}));
+
+vi.mock('../ui/TrayManager', () => ({
+  TrayManager: vi.fn().mockImplementation(() => ({
+    initializeTrays: vi.fn(),
+    destroyAll: vi.fn(),
+  })),
+}));
+
+vi.mock('@/utils/next-electron-rsc', () => ({
+  createHandler: vi.fn(() => ({
+    createInterceptor: vi.fn(),
+    registerCustomHandler: vi.fn(),
+  })),
+}));
+
+// Mock controllers and services
+vi.mock('../../controllers/*Ctr.ts', () => ({}));
+vi.mock('../../services/*Srv.ts', () => ({}));
+
+// Import after mocks are set up
+import { App } from '../App';
+
+describe('App - Database Lock Cleanup', () => {
+  let appInstance: App;
+  let mockLockPath: string;
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+
+    // Mock glob imports to return empty arrays
+    (import.meta as any).glob = vi.fn(() => ({}));
+
+    mockLockPath = join('/mock/storage/path', LOCAL_DATABASE_DIR) + '.lock';
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  describe('bootstrap - database lock cleanup', () => {
+    it('should remove stale lock file if it exists during bootstrap', async () => {
+      // Setup: simulate existing lock file
+      vi.mocked(pathExistsSync).mockReturnValue(true);
+      vi.mocked(remove).mockResolvedValue(undefined);
+
+      // Create app instance
+      appInstance = new App();
+
+      // Call bootstrap which should trigger cleanup
+      await appInstance.bootstrap();
+
+      // Verify: lock file check was called
+      expect(pathExistsSync).toHaveBeenCalledWith(mockLockPath);
+
+      // Verify: lock file was removed
+      expect(remove).toHaveBeenCalledWith(mockLockPath);
+    });
+
+    it('should not attempt to remove lock file if it does not exist', async () => {
+      // Setup: no lock file exists
+      vi.mocked(pathExistsSync).mockReturnValue(false);
+
+      // Create app instance
+      appInstance = new App();
+
+      // Call bootstrap
+      await appInstance.bootstrap();
+
+      // Verify: lock file check was called
+      expect(pathExistsSync).toHaveBeenCalledWith(mockLockPath);
+
+      // Verify: remove was NOT called since file doesn't exist
+      expect(remove).not.toHaveBeenCalled();
+    });
+
+    it('should continue bootstrap even if lock cleanup fails', async () => {
+      // Setup: simulate lock file exists but cleanup fails
+      vi.mocked(pathExistsSync).mockReturnValue(true);
+      vi.mocked(remove).mockRejectedValue(new Error('Permission denied'));
+
+      // Create app instance
+      appInstance = new App();
+
+      // Bootstrap should not throw even if cleanup fails
+      await expect(appInstance.bootstrap()).resolves.not.toThrow();
+
+      // Verify: cleanup was attempted
+      expect(pathExistsSync).toHaveBeenCalledWith(mockLockPath);
+      expect(remove).toHaveBeenCalledWith(mockLockPath);
+    });
+
+    it('should clean up lock file before starting IPC server', async () => {
+      // Setup
+      vi.mocked(pathExistsSync).mockReturnValue(true);
+      const callOrder: string[] = [];
+
+      vi.mocked(remove).mockImplementation(async () => {
+        callOrder.push('remove');
+      });
+
+      // Mock IPC server start to track call order
+      const { ElectronIPCServer } = await import('@lobechat/electron-server-ipc');
+      const mockStart = vi.fn().mockImplementation(() => {
+        callOrder.push('ipcServer.start');
+        return Promise.resolve();
+      });
+
+      vi.mocked(ElectronIPCServer).mockImplementation(
+        () =>
+          ({
+            start: mockStart,
+          }) as any,
+      );
+
+      // Create app instance and bootstrap
+      appInstance = new App();
+      await appInstance.bootstrap();
+
+      // Verify: cleanup happens before IPC server starts
+      expect(callOrder).toEqual(['remove', 'ipcServer.start']);
+    });
+  });
+
+  describe('appStoragePath', () => {
+    it('should return storage path from store manager', () => {
+      appInstance = new App();
+
+      const storagePath = appInstance.appStoragePath;
+
+      expect(storagePath).toBe('/mock/storage/path');
+    });
+  });
+});

